internal/forms/forms.go

```diff
@@ -705,6 +709,108 @@ func (m *Manager) findAbsPathForTemplatePath(tmplPath string) (string, error) {
 	return retVal, nil
 }
 
+func (m *Manager) gpsPos() (gpsd.Position, error) {
+	// return current GPS Position
+
+	addr := m.config.GPSd.Addr
+	if addr != "" {
+		conn, err := gpsd.Dial(addr)
+		if err != nil {
+			log.Printf("GPSd daemon: %s", err)
+		}
+		defer conn.Close()
+
+		conn.Watch(true)
+
+		log.Println("Waiting for position from GPSd...")
+		pos, err := conn.NextPosTimeout(3 * time.Second) // TODO: configurable
+		if err != nil {
+			log.Printf("GPSd: %s", err)
+		} else {
+			log.Printf("GPSd: %f %f", pos.Lat, pos.Lon)
+		}
+		return pos, err
+	} else {
+		return gpsd.Position{}, errors.New("GPSd: not configured.")
+	}
+}
+
+type gpsStyle int
+
+const (
+	// documentation: https://www.winlink.org/sites/default/files/RMSE_FORMS/insertion_tags.zip
+	signedDecimal gpsStyle = iota // 41.1234 -73.4567
+	decimal                       // 46.3795N 121.5835W
+	degreeMinute                  // 46-22.77N 121-35.01W
+)
+
+func gpsFmt(style gpsStyle, pos gpsd.Position) string {
+	var northing string
+	var easting string
+	var latDegrees int
+	var latMinutes float64
+	var lonDegrees int
+	var lonMinutes float64
```

Comment:
It's preferable to define variables as local as possible. I think this is a symptom of the complexity introduced by accepting the style parameter instead of having multiple implementing functions for the different styles.

PS: When declaring lots of variables like this, it is a bit cleaner to use this syntax:

```go
var (
   northing string
   easting string
   latDegrees int
   latMinutes float64
   lonDegrees int
   lonMinutes float64
)
```